src/NCTablePad.cc

```diff
@@ -30,6 +30,53 @@
 #include <limits.h>
 
 
+void
+NCTableSortDefault::sort ( std::vector<NCTableLine *>::iterator itemsBegin,
+			   std::vector<NCTableLine *>::iterator itemsEnd )
+{
+    std::sort ( itemsBegin, itemsEnd, Compare(getColumn(), isReverse()) );
+}
+
+
+bool
+NCTableSortDefault::Compare::operator() ( const NCTableLine * first,
+					  const NCTableLine * second ) const
+{
+    std::wstring w1 = smartSortKey( first );
+    std::wstring w2 = smartSortKey( second );
+    wchar_t * endptr1 = 0;
+    wchar_t * endptr2 = 0;
+
+    long long number1 = std::wcstoll( w1.c_str(), &endptr1, 10 );
+    long long number2 = std::wcstoll( w2.c_str(), &endptr2, 10 );
+
+    if ( *endptr1 == L'\0' && *endptr2 == L'\0' )
```

Comment:
I see one issue with this kind of conversion. If there is "" and "-5" then order is "-5" and then "0" as both return empty endptr and one set value to 0  due to error and second set -5.
If it is expected sorting, then please document it as it is not obvious from first sign. If it is not expected, then man page suggest this:
```
Since 0, {LONG_MIN} or {LLONG_MIN} and {LONG_MAX} or {LLONG_MAX} are returned on error and are also valid returns on success, an application wishing to check for error situations should set errno to 0, then call wcstol() or
       wcstoll(), then check errno.
```

Reply:
Ah, I see it is also original behavior, which is a kind of buggy. So it would be nice to fix it, but not mandatory.

Reply:
I have now improved the function to use C++11 stoll() function. I have also changed it so that all integers are placed before strings as it is done in libyui-qt.